common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -86,6 +87,52 @@ def default(self, obj):
         return json.JSONEncoder.default(self, obj)
 
 
+class WarningFieldSingleValueMixin(JSONField):
+    """
+    Wraps around the actual value of another xmodule.field.Field.
+    This mixin must be used with a field that does not hard-code
+    a set of values, as it uses the "_values" field to communicate
+    warnings to the front-end
+
+    The condition should be of the form "<fieldname> <op> <value>"
+    similar to a SQL where clause. Currently supported "op"
+    is only ">", and can be implemented in a need-by-need basis
+    on the front-end. "fieldname" is a CapaField.
+    """
+    def __init__(self, warning=None, warning_condition=None, **kwargs):
+        super(WarningFieldSingleValueMixin, self).__init__(**kwargs)
+        self._values = {
+            "original_value": self._values,
+            "warning": warning,
+            "warning_condition": warning_condition
+        }
+
+    def from_json(self, value):
+        """
+        Override native method to extract "original_value" field
+        from natively stored Python object
+        """
+        _from_json = super(WarningFieldSingleValueMixin, self).from_json
+        # check whether we have for our mixin an original "from_json"
+        # function from inheriting from an Integer, String, Dict, or
+        # other field type
+        if _from_json is not None and type(_from_json) == MethodType:
+            true_value = _from_json(value)
+            self._values["original_value"] = true_value
+        else:
+            raise TypeError('Value stored in a WarningFieldSingleValueMixin must be None or a dict, found %s' % type(value))
+        return self._values["original_value"]
+
+
+class IntegerWithWarningField(WarningFieldSingleValueMixin, Integer):
```

Comment:
You're inheriting from both `JSONField` and `Integer`, which are two different types of xblock fields.  This seems likely to cause some issues with inherited behavior, no?